### PR TITLE
DB-11112 Kafka Maintenance to Delete Specified Topics

### PR DESCRIPTION
**Pull request by jpanko1**

### CODE REVIEW

**Comment by jpanko1** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

File: splice_spark2/src/main/java/com/splicemachine/nsds/kafka/KafkaMaintenance.java [hunk 1]
```diff
@@ -96,19 +100,43 @@ public static void deleteOldTopics(String kafkaServers, String dataFileName, lon
...
                 ageCutoffMinutes = Long.parseLong( args[2] );
             }
-            KafkaMaintenance.deleteOldTopics( args[0] , args[1] , ageCutoffMinutes );
+            List<String> prefixToDelete = null;
+            if( args.length >= 4 && !args[3].isEmpty() ) {
+                prefixToDelete = Arrays.asList( args[3].split(",") );
+            }
+            List<String> prefixToKeep = null;
+            if( args.length >= 5 && !args[4].isEmpty() ) {
+                if( !args[3].isEmpty() && !args[4].isEmpty() ) {
```
Comment (msirek) [inline splice_spark2/src/main/java/com/splicemachine/nsds/kafka/KafkaMaintenance.java:129]: Remove redundant condition in inner if block: !args[4].isEmpty()

**Review by msirek — APPROVED** [review]

**Review by jyuanca — APPROVED** [review]

File: splice_spark2/src/main/java/com/splicemachine/nsds/kafka/KafkaMaintenance.java (hunk 1 above)
Comment (jpanko1) [reply splice_spark2/src/main/java/com/splicemachine/nsds/kafka/KafkaMaintenance.java]: Fixed in commit https://github.com/splicemachine/spliceengine/pull/4991/commits/d04691e8f2c188cfdb70f38c43a4b57835318f8f

**Comment by jpanko1** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Review by carolp-503 — APPROVED** [review]
+1

**Pull request merged**